modules/core/include/opencv2/core/mat.inl.hpp

```diff
@@ -668,8 +668,9 @@ void Mat::addref()
 
 inline void Mat::release()
 {
-    if( u && CV_XADD(&u->refcount, -1) == 1 )
-        deallocate();
+    if (u)
+        CV_XADD(&u->refcount, -1);
+    deallocate();
```

Comment:
```
if (u)
{
    CV_XADD(&u->refcount, -1);
    deallocate();
}
```

BTW, This header contains inlined functions. It is not safe to change this without client code recompilation.